app/src/test/java/org/oppia/android/app/translation/AppLanguageWatcherMixinTest.kt

```diff
@@ -206,15 +206,41 @@ class AppLanguageWatcherMixinTest {
   }
 
   @Test
-  fun testMixin_initialized_withAppLanguageChange_newLanguage_recreatesActivity() {
```

Comment:
Hi Ben updated to initialize profiles before running this test to resolve the issue of null profileId but ProfileManagmentController is always returning null this seems to be failing from this line when profileInitialization is in progress https://github.com/oppia/oppia-android/blob/711d5f6f67e5a0f5ca12aedc1f67a288a78ba1f4/domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt#LL707C75-L707C75  currentProfileId will not be updated as expected hence null will always be returned please advice. Thanks.

Reply:
So the problem was actually two things:
1. Profile initialization & login needs to occur before the activity is opened, so the test needed to be switched over to the "launch-after-setup" pattern that we use elsewhere.
2. I used ProfileTestHelper for easier initialization & login, but it had some unreliability in how it timed things so I made some changes to make that more robust.

Because of the two changes above, I was able to re-add the original two tests that were removed with some minor changes. These will be incorporated in my follow-up PR.